### PR TITLE
[WIP] Structure revamp of Serverless docs

### PR DESCRIPTION
**Pull request by abrennan89**
Major structural changes.

1. Improved getting started intro content. Next steps changed to additional resources.
1. New getting_started folder; installation folder removed, topics now split between main serverless root folder and getting_started.
1. Edited topicmap to reflect moved topics.
1. Edits / updates to removing serverless section (later todo: split last part of this into its own module re deleting CRDs).
1. Updated xrefs for moved files.
1. Improvements to content based on last minimalism review.
1. Update code blocks to use new formal, [source,x] etc.
1. Moved subscription module content to assembly, there were xrefs in the module for some reason.
1. Sinkbinding moved to event_sources
1. Metering, tracing and HA moved to knative_serving
1. supported config line with link got its own module so that it can be reused
1. moved arch sections into getting started before sections for installing each component respectively
1. New about serverless section; previously removed due to minimalism guidelines re using this as a heading, but it's used a lot in other places, so re-adding it until there's a viable alt

To do:

- [x] Follow up with sheet updates to have links rerouted to new locations.
- [x] CP to all required branches and resolve any conflicts.
  
  - #25221 CP to 4.3
  - #25262 CP to 4.4
  - #25263 CP to 4.5
  - #25264 CP to 4.6

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@abrennan89 ping me when this merges.

**Review by cardil — APPROVED** [review]
/lgtm

From QE. I didn't saw any obvious problems.

File: _topic_map.yml
Comment (bergerhoffer) [inline _topic_map.yml:2406]: I know it was like this before it was moved, but "Release Notes" really should be "Release notes". Same w/ the name of the assembly.

File: serverless/knative_eventing/serverless-kn-trigger.adoc
Comment (bergerhoffer) [inline serverless/knative_eventing/serverless-kn-trigger.adoc:74]: I see you adding [source,terminal] a few places, but in looking through the entire serverless book, there are a ton that don't have this yet. I might suggest to not worry about that in this PR, and then open a PR after this is merged to add those in.

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (bergerhoffer) [inline serverless/knative_eventing/serverless-subscriptions.adoc:13]: We have a `technology-preview.adoc` file that we include so that we use consistent wording for TP features. Suggest following that here (just search the repo for that file and you'll see how it's used).

File: modules/serverless-supported-configs.adoc
Comment (bergerhoffer) [inline modules/serverless-supported-configs.adoc:1]: This file is missing the metadata that says which assemblies it's used in.

File: modules/serverless-supported-configs.adoc
Comment (bergerhoffer) [inline modules/serverless-supported-configs.adoc:2]: Use sentence case

File: modules/serverless-supported-configs.adoc
Comment (bergerhoffer) [inline modules/serverless-supported-configs.adoc:4]: could you simplify to 

```
The set of supported features, configurations, and integrations for all versions of {ServerlessProductName} are available at the link:https://access.redhat.com/articles/4912821[Supported Configurations page].
```
Also maybe "at the" -> "on the" ?

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:22]: Suggest making all these subsections into modules and then including them from the assembly.

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:9]: cloud native -> cloud-native?

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:12]: The way this is worded, I'm not sure whether the second part is saying that Knative project enables an enterprise-grade platform, or if it's supposed to mean that OpenShift Serverless does.

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:15]: Should this say "the need for complex infrastructure to be set up before..." ?

Or maybe "the need for a complex infrastructure setup before..." ?

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:17]: out-of-the-box

Comment (bergerhoffer) [reply serverless/about-serverless.adoc:17]: Also this sentence is a little long/complex. I'd consider breaking it up a little if possible.

File: serverless/about-serverless.adoc
Comment (bergerhoffer) [inline serverless/about-serverless.adoc:39]: * Should "Revision" and "Configuration" be lowercase? (Same elsewhere in the PR)
* user created -> user-created?
* Probably lose the comma before custom resource

**Review by bergerhoffer — COMMENTED** [review]
I didn't finish a whole review, but sharing the feedback so far

File: serverless/knative_eventing/serverless-kn-trigger.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-kn-trigger.adoc:21]: s/procedure below/following procedure/
**below** _adj_
Do not use to indicate a relative location in a document, as in “the information below.”

File: serverless/getting_started/installing-openshift-serverless.adoc
Comment (bobfuru) [inline serverless/getting_started/installing-openshift-serverless.adoc:24]: s/40GB/40 GB/

File: serverless/getting_started/installing-openshift-serverless.adoc
Comment (bobfuru) [inline serverless/getting_started/installing-openshift-serverless.adoc:49]: s/96GB/96 GB/
For consistency with this doc and ISG, could you change this also?

File: serverless/getting_started/installing-openshift-serverless.adoc
Comment (bobfuru) [inline serverless/getting_started/installing-openshift-serverless.adoc:54]: s/200MB - 2GB/200 MB - 2 GB/

File: serverless/about-serverless.adoc
Comment (bobfuru) [inline serverless/about-serverless.adoc:18]: s/fine tune/fine-tune/

File: serverless/getting_started/installing-knative-serving.adoc
Comment (bobfuru) [inline serverless/getting_started/installing-knative-serving.adoc:13]: You have "Custom Resource Definition (CRD)" elsewhere in this PR, so I'd capitalize here for consistency.

File: serverless/getting_started/installing-openshift-serverless.adoc
Comment (bobfuru) [inline serverless/getting_started/installing-openshift-serverless.adoc:9]: s/prerequisites and installation/prerequisites, and installation/

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-subscriptions.adoc:19]: s/wish/want/

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-subscriptions.adoc:24]: s/service, by/service by/

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (bobfuru) [inline serverless/knative_eventing/serverless-subscriptions.adoc:53]: Not sure if this would need an ack from Product Experience/Support because it is an external link?

File: serverless/knative_serving/serverless-metering.adoc
Comment (bobfuru) [inline serverless/knative_serving/serverless-metering.adoc:11]: To me, "For more information about metering on OpenShift" sounds odd. Maybe "...about {product-title} metering" or "about using metering on {product-title}..."?

File: serverless/serverless-support.adoc
Comment (bobfuru) [inline serverless/serverless-support.adoc:11]: Just curious why you removed the `_{context}` separator here? I see the variable is still defined in the assembly (line 4). (I confess that I don't fully understand how we use this variable.)

**Review by bobfuru — COMMENTED** [review]
Overall, LGTM! I added a few minor suggestions in addition to what Andrea already added earlier in her review.

File: serverless/knative_eventing/serverless-kn-trigger.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-kn-trigger.adoc:74]: Cool, I'll leave the ones I've added in this PR as are though if that's OK? :slightly_smiling_face: Or do I need to revert them?

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-subscriptions.adoc:13]: I actually use it somewhere else, but this note is specifically to call out the thing about InMemoryChannel rather than the TP status. I'll remove the first line to make this the focus. Thanks.

File: modules/serverless-supported-configs.adoc
Comment (abrennan89) [reply modules/serverless-supported-configs.adoc:1]: Oops, added thanks.

File: serverless/about-serverless.adoc
Comment (abrennan89) [reply serverless/about-serverless.adoc:39]: I don't think so re lowercase - It's kind of like service vs. Service, these are referring to a specific type of object as opposed to the general concept of "configurations" or "revisions" etc, so it makes sense to me to give them caps?

This should probably be part of the overall discussions happening around Kubernetes objects, what to capitalize etc, so any guidance is appreciated, I'm sort of waiting to see what the OpenShift guidelines end up being before I recommend anything specific for Serverless or Knative :slightly_smiling_face: 

Maybe a follow up issue/PR to track adding these specific resource types to the glossary would be worthwhile?

File: serverless/getting_started/installing-knative-serving.adoc
Comment (abrennan89) [reply serverless/getting_started/installing-knative-serving.adoc:13]: I think it should actually be `CustomResourceDefinition` according to https://github.com/openshift/openshift-docs/pull/22440 - I forgot about this tbh.

It doesn't seem to be consistently applied in a bunch of places across docs yet though, most seem to use "Custom Resource Definition (CRD)" or similar, so I will use this for now and put in a separate PR later to update all the instances to `CustomResourceDefinition` if that's OK?

File: serverless/knative_eventing/serverless-subscriptions.adoc
Comment (abrennan89) [reply serverless/knative_eventing/serverless-subscriptions.adoc:53]: Removed for now.

File: serverless/serverless-support.adoc
Comment (abrennan89) [reply serverless/serverless-support.adoc:11]: Afaik it's just needed in modules, not assembly ID headings, but I could be wrong?

**Comment by abrennan89** [comment]
Added changes from reviews except for the couple of comments I've left @bobfuru @bergerhoffer - if there's no other feedback this should be OK to merge.

I've added links in the description to the PRs for CPing all these changes to 4.3 - 4.6 branches and dealing with any conflicts.
@cardil these other PRs should be fine to QE review once we have confirmation that there are no more peer review changes.

Thanks folks!

**Review by bobfuru — COMMENTED** [review]
Updates LGTM - please squash commits. And if possible, let's wait until @bergerhoffer can also re-review. Thanks!

**Comment by vikram-redhat** [comment]
Hey @abrennan89 - does this restructure undo any of the changes that we recently did for which we added some redirects?

File: serverless/serverless-support.adoc
Comment (vikram-redhat) [reply serverless/serverless-support.adoc:11]: The templates for mod docs don't specify the need to have an anchor for headings in assemblies. @openshift/team-documentation - do we need to specify this?

Comment (bergerhoffer) [reply serverless/serverless-support.adoc:11]: Nope, the _{context} part should only be needed for modules.

File: serverless/serverless-release-notes.adoc
Comment (bergerhoffer) [inline serverless/serverless-release-notes.adoc:3]: Sorry I mislead you with that other comment, this should technically be "release notes" both lowered.

**Comment by bergerhoffer** [comment]
@abrennan89 I re-reviewed and had 2 comments that might have been collapsed that you didn't ack. It's fine if you chose to not doing anything for those, but just wanted to make sure you didn't miss them just because they were collapsed.

I had just one last thing, and then other than @vikram-redhat's question and needing to squash, the rest LGTM.

**Pull request closed without merging**
